src/Umbraco.Infrastructure/Intall/InstallHelper.cs

```diff
@@ -22,32 +22,38 @@ public sealed class InstallHelper
     {
         private static HttpClient _httpClient;
         private readonly DatabaseBuilder _databaseBuilder;
-        private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly ILogger _logger;
         private readonly IGlobalSettings _globalSettings;
         private readonly IUmbracoVersion _umbracoVersion;
         private readonly IConnectionStrings _connectionStrings;
         private readonly IInstallationService _installationService;
         private readonly ICookieManager _cookieManager;
+        private readonly IUserAgentProvider _userAgentProvider;
+        private readonly Lazy<ISqlContext> _sqlContext;
+        private readonly IJsonSerializer _jsonSerializer;
         private InstallationType? _installationType;
 
-        public InstallHelper(IHttpContextAccessor httpContextAccessor,
-            DatabaseBuilder databaseBuilder,
+        public InstallHelper(DatabaseBuilder databaseBuilder,
             ILogger logger,
             IGlobalSettings globalSettings,
             IUmbracoVersion umbracoVersion,
             IConnectionStrings connectionStrings,
             IInstallationService installationService,
-            ICookieManager cookieManager)
+            ICookieManager cookieManager,
+            IUserAgentProvider userAgentProvider,
+            Lazy<ISqlContext> sqlContext,
```

Comment:
Introducing a new circular ref? aha, i see it's in order to get rid of the `Current` usage. Perhaps we should add TODO notes about these things to see if we can un-circular ref these.

Reply:
In general, I do not introduce new circular refs, but the circles become more explicit by the move into constructor injection :)

I changed this case to inject IUmbracoDatabaseFactory instead, and the call SqlContext on that dependency. I'm not a big fan of that solution either, But as long as all of our interfaces and classes has such big responsibility, the cicular refs are hard to avoid.